### DISCUSSION TITLE
Repository test failures

### DISCUSSION THREAD

**Discussion by zoobert** [open-telemetry/opentelemetry-java-instrumentation#8504] [Q&A]
I am trying to start contributing to this repository, but cannot run a full build on my system. There were a few projects where there were test failures. I had to exclude the test task for the following projects to complete the build:

- instrumentation:oshi:library
- instrumentation:aws-sdk:aws-sdk-1.11:javaagent
- instrumentation:couchbase:couchbase-3.1:javaagent

I've looked at the failures and did not see anything obvious, but I wanted to pose a question before I dug into each of the failures or posted the failures here. Is completing a full build on my system necessary when I only intend to change a subproject? I know it is good form, but spending time debugging/understanding the build failures limits my ability to contribute small changes. To be clear, I intend to submit a single dependency change to fix this: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8474

My system:
2022 16" MBP/M1 Max
32 GB
Ventura 13.3.1 (a)

**Comment by jkwatson** [comment]
moving to the instrumentation repository

**Comment by laurit** [comment]
Full build isn't necessary. Before you submit your pr it is a good idea to run build on the module you changed to catch spotless and checkstyle errors, if you don't it is also not a big deal, you can always push another commit.
As I haven't tried M1 I can't say for sure whether it is possible to get all the tests passing, I guess `oshi` is a good candidate for something that could fail on M1. Usually the failures I get during local builds are related to running too many tests in parallel. You can run the build with `--continue` and rerun it at the end to avoid having to restart the build on each failure. Using `--max-workers=...` can also improve the test run stability.
